### DISCUSSION TITLE
Queuing and keeping messages while broker unavailable

### DISCUSSION THREAD

**Discussion by EkimEvgeny** [Koenkk/zigbee2mqtt#6825] [General]
Does any setting exist that allows zigbee2mqtt to keep messages while MQTT broker unavailable and send them when connection restored?
I need it for a project, where the internet can be unstable, and need to keep all sensor measurements anyway.
If no - can I add this feature to zigbee2mqtt?

**Comment by Koenkk** [comment]
You can set `retain: true` for those device, e.g. https://www.zigbee2mqtt.io/information/configuration.html#example 

http://www.steves-internet-guide.com/mqtt-retained-messages-example/
